src/node_report_utils.cc

```diff
@@ -82,6 +83,40 @@ static void ReportEndpoints(uv_handle_t* h, JSONWriter* writer) {
   ReportEndpoint(h, rc == 0 ? addr : nullptr, "remoteEndpoint", writer);
 }
 
+// Utility function to format libuv pipe information.
+static void ReportPipeEndpoints(uv_handle_t* h, JSONWriter* writer) {
+  uv_any_handle* handle = reinterpret_cast<uv_any_handle*>(h);
+  MaybeStackBuffer<char> buffer;
+  size_t buffer_size = buffer.capacity();
+  int rc = -1;
+
+  rc = uv_pipe_getsockname(&handle->pipe, buffer.out(), &buffer_size);
+  if (rc == UV_ENOBUFS) {
+    // Buffer is not large enough, reallocate to the updated buffer_size
+    // and fetch the value again.
+    buffer.AllocateSufficientStorage(buffer_size);
```

Comment:
Is this (`AllocateSufficientStorage`) guaranteed to pass always? Should we abandon if it fails? (same below too)

Reply:
Aborting here would be different behaviour to `ReportPath()` in this file, which would continue best effort to produce a report if memory allocation fails.

Reply:
sorry if I confused with the word `abandon`. Given some use cases under which report kicks-in (abnormal program conditions which include memory-crunch situations as well), going with the best effort is more ideal here, than hard abort IMO. I mean `abandon` to indicate give up the current attempt (the section under work) and move on.

Reply:
Thanks, noticed that the `MallocedBuffer` can replate the `MaybeStackBuffer` as the `MallocedBuffer` is using the unchecked type of realloc.

Updated.